.github/workflows/pull_request_tests.yaml

```diff
@@ -0,0 +1,40 @@
+name: Test Action in CI
+
+on:
+  pull_request:
+    branches:
+      - master
+
+env:
+  CLANG_TIDY_VERSION: 17.0.1
+  CMAKE_VERSION: 3.21.3
+  # cannot use > 3.11 due to https://github.com/ssciwr/clang-tidy-wheel/issues/66
```

Comment:
Aha, I guess this is due to using `pip` to install `clang-tidy`

Reply:
Correct.